### PR TITLE
Use DCDO key to hide volunteers from map until next Hour of Code

### PR DESCRIPTION
**Pull request by breville**
This hides volunteers, who said they want to unsubscribe until the next Hour of Code, from the volunteer map.  

Instead of reverting the code to how it was before the last Hour of Code, this sets up a DCDO key called "hoc_mode" which will make these volunteers start appearing again when it becomes "actual-hoc".  

The hope is that we can start using this DCDO key to transition the site through a variety of modes.

### CODE REVIEW

File: pegasus/forms/volunteer_engineer_submission_2015.rb [hunk 1]
```diff
@@ -108,9 +108,15 @@ def self.process(data)
   end
 
   def self.solr_query(params)
-    # Remove and add UNSUBSCRIBE_HOC before and after each Hour of Code.
     query = "kind_s:\"#{name}\" && allow_contact_b:true && -unsubscribed_s:\"#{UNSUBSCRIBE_FOREVER}\""
 
+    # UNSUBSCRIBE_HOC means a volunteer said "I want to unsubscribe until the next Hour of Code".
+    # We don't want them to be getting volunteer requests until then.  So, if we're not currently
+    # in Hour of Code, don't show that volunteer, and do that by including UNSUBSCRIBE_HOC here.
```
Comment (ashercodeorg) [inline pegasus/forms/volunteer_engineer_submission_2015.rb:115]: Don't we want to open up requests slightly before CS Education Week? Do we need "pre-hoc" and "actual-hoc" modes?

Comment (breville) [reply pegasus/forms/volunteer_engineer_submission_2015.rb:115]: Quite possibly... @tanyaparker?

Comment (tanyaparker) [reply pegasus/forms/volunteer_engineer_submission_2015.rb:115]: Yes, volunteers should be in the map in the preceding season so teachers have time to coordinate with them before actual HOC. I have usually communicated HOC season to be Oct-Dec and we kick off volunteers with a webinar in early October reminding them that teachers are going to start contacting them again.

Are we starting with the list of hoc modes that I suggested? If so where is that documented/how do I find out what mode is currently set? Is it through a UI for admins?

Comment (tanyaparker) [reply pegasus/forms/volunteer_engineer_submission_2015.rb:115]: Thanks for the wiki link! Yes, HOC only volunteers should be on the map during both `soon-hoc` and `actual-hoc` modes.

Comment (breville) [reply pegasus/forms/volunteer_engineer_submission_2015.rb:115]: Updated to match these two modes.  Will discuss with infrastructure folks how to best store & expose this mode.

**Pull request merged**